python/cudf/cudf/_libxx/sort.pyx

```diff
@@ -12,10 +12,96 @@ from cudf._libxx.move cimport move
 
 from cudf._libxx.cpp.column.column cimport column
 from cudf._libxx.cpp.table.table_view cimport table_view
-from cudf._libxx.cpp.sort cimport (
-    sorted_order, lower_bound, upper_bound
-)
-cimport cudf._libxx.cpp.types as libcudf_types
+from cudf._libxx.cpp.search cimport lower_bound, upper_bound
+from cudf._libxx.cpp.sorting cimport sorted_order, is_sorted as cpp_is_sorted
+from cudf._libxx.cpp.types cimport order, null_order
+
+
+def is_sorted(
+    Table source_table, object col_order=None, object null_prec=None
+):
+    """
+    Checks whether the rows of a `table` are sorted in lexicographical order.
+
+    Parameters
+    ----------
+    source_table : Table
+        Table whose columns are to be checked for sort order
+    col_order : None or list-like of booleans
+        None or list-like of boolean values indicating expected sort order of
+        each column. If list-like, size of list-like must be len(columns). If
+        None, all columns expected sort order is set to ascending. False (0) -
+        ascending, True (1) - descending.
+    null_prec : None or list-like of booleans
+        None or list-like of boolean values indicating desired order of nulls
+        compared to other elements. If list-like, size of list-like must be
+        len(columns). If None, null order is set to before. False (0) - before,
+        True (1) - after.
+
+    Returns
+    -------
+    returns : boolean
+        Returns True, if sorted as expected by ``col_order`` and ``null_prec``,
+        False otherwise.
+    """
+
+    cdef vector[order] column_order
+    cdef vector[null_order] null_precedence
+
+    if col_order is None:
+        column_order = vector[order](
+            source_table._num_columns, order.ASCENDING
+        )
+    elif pd.api.types.is_list_like(col_order):
+        if len(col_order) != source_table._num_columns:
+            raise ValueError(
+                f"Expected a list-like of length {source_table._num_columns}, "
+                f"got length {len(col_order)} for `col_order`"
+            )
+        column_order = vector[order](
+            source_table._num_columns, order.ASCENDING
+        )
+        for idx, val in enumerate(col_order):
+            if val:
```

Comment:
nit: Pandas uses `ascending=True` as default, and this functionality propagates publicly. Here, we use `True` for descending. Not directly related, but I dunno if we want to match the Pandas behavior.

Reply:
It probably makes sense to rename the parameters to `ascending` and `null_before` and handle it as such. Will make the changes.

Reply:
Done.